### ISSUE TITLE
where is yolov5 paper?

### ISSUE THREAD

**Issue by eeric** [ultralytics/yolov5#1333]
## ❔Question

where is paper

**Comment by glenn-jocher** [comment]
@eeric hi thanks for asking! We aim to publish a short paper to arxiv by the end of 2020. In the meantime please see https://doi.org/10.5281/zenodo.3908559 for DOI to cite this repo directory, and where you can also find bibtex and other exportable citation formats.

[![DOI](https://zenodo.org/badge/264818686.svg)](https://zenodo.org/badge/latestdoi/264818686)

**Comment by hujunchao** [comment]
Looking forward to meeting YOLOv5 paper!

**Comment by NavneetSajwan** [comment]
Is the paper out yet?

**Comment by selenecodes** [comment]
Any updates on this?

**Comment by ric-jon** [comment]
This probably shouldn't be closed yet.

**Comment by robertokcanale** [comment]
Any updates?

**Comment by vikxoxo** [comment]
Any paper or detailed doc, illustrating the architecture

**Comment by glenn-jocher** [comment]
@vikxoxo @robertokcanale it's not as good as a paper, but you can visualize the architectures now after our Tensorboard fix in PR https://github.com/ultralytics/yolov5/pull/2758. Hopefully paper will follow soon.

This is a YOLOv5s model displayed in TensorBoard. You can see the Detect() layer merging the 3 layers into a single output for example, and everything appears to work and visualize correctly. You can get this by uncommenting the tb_writer.add_graph() lines 333 and 335 in train.py:
https://github.com/ultralytics/yolov5/blob/0f395b3e3bccbc019ab3d1cbd41303a5b50dc0f0/train.py#L333-L335

<img width="893" alt="Screenshot 2021-04-11 at 01 10 09" src="https://user-images.githubusercontent.com/26833433/114286928-349bd600-9a63-11eb-941f-7139ee6cd602.png">

**Comment by robertokcanale** [comment]
@glenn-jocher Thank you, I'll try it out. This looks good enough for my Master Thesis! But as I might also plan a publication on this, a paper would be of great help, can't wait for it, hoping reviewers accept it smoothly!

**Comment by Muhtadir** [comment]
@robertokcanale same here sir.

**Comment by cenkbircanoglu** [comment]
+1

**Comment by MaKaNu** [comment]
It is 2021 and still no paper. With no paper this Work is pretty much useless from a scientific point of view.

**Comment by bakirsw** [comment]
I there any chance that it will be published this month?

**Comment by ksasso1028** [comment]
a paper would be really nice. But the fact that you have the architecture in pure pytorch is very helpful. Anyone with enough DL exp could trace this arch out and break down the types of layers you are using. I wish every paper that came out had a Pytorch implementation

**Comment by suryagutta** [comment]
A paper would be beneficial!

**Comment by rossGardiner** [comment]
Will there be a paper for Yolov5 before the end of the year? In a research context, I can't really commit to using Yolov5 if there is no paper describing inner workings...

**Comment by zxsitu** [comment]
@rossGardiner @glenn-jocher Same feeling as you. I am also a bit worried about using yolov5 for my article at the moment, although it is an excellent engineering project, the lack of an introductory article is really hard to convince the reviewers of the journal.

**Comment by besmaguesmi** [comment]
Hey, is the Yolov5 paper out yet?

**Comment by glenn-jocher** [comment]
@besmaGuesmi we are targeting a paper release by the start of PyTorch Dev day, December 1st. 

If the YOLOv5 paper is not published by then I will eat my hat.

**Comment by besmaguesmi** [comment]
@glenn-jocher sounds great! all the best.

**Comment by Keiku** [comment]
The next release may be less confusing if the paper and code are released at the same time. I'm looking forward to it.

**Comment by Makki93** [comment]
@glenn-jocher any news about the paper yet? :)
1st of December is soon and I wanted to reference it in my thesis with the due date end of December.
Also, I don't want to see you eating a hat.

**Comment by muhammedakyuzlu** [comment]
Today is the premise day @glenn-jocher

**Comment by maxl-13** [comment]
everyone is still waiting :)

**Comment by fazer125** [comment]
@glenn-jocher 
Where is paper?
Will you eat your hat?

**Comment by felixdittrich92** [comment]
@glenn-jocher any update ? :)

**Comment by gkm0120** [comment]
@glenn-jocher Now on December 2, 2021, there are still no papers :)

**Comment by johnmarkacala** [comment]
> @vikxoxo @robertokcanale it's not as good as a paper, but you can visualize the architectures now after our Tensorboard fix in PR #2758. Hopefully paper will follow soon.
> 
> This is a YOLOv5s model displayed in TensorBoard. You can see the Detect() layer merging the 3 layers into a single output for example, and everything appears to work and visualize correctly. You can get this by uncommenting the tb_writer.add_graph() lines 333 and 335 in train.py:
> 
> https://github.com/ultralytics/yolov5/blob/0f395b3e3bccbc019ab3d1cbd41303a5b50dc0f0/train.py#L333-L335
> 
> <img alt="Screenshot 2021-04-11 at 01 10 09" width="893" src="https://user-images.githubusercontent.com/26833433/114286928-349bd600-9a63-11eb-941f-7139ee6cd602.png">

I cant seem to find this in train.py anymore. Any other recommendations?

**Comment by Makki93** [comment]
There is this paper:
TPH-YOLOv5: Improved YOLOv5 Based on Transformer Prediction Head for Object Detection on Drone-captured Scenarios
https://arxiv.org/pdf/2108.11539.pdf

As well as this thesis:
from https://www.theseus.fi/bitstream/handle/10024/452552/Do_Thuan.pdf

Quote:
"
5TH GENERATION OF YOLO
A month after YOLOv4 was released, researcher Glenn and his team published a new version of 
the YOLO family, called YOLOv5 (Jocher, 2020). Glenn Jocher is a researcher and CEO of 
Ultralystics LLC. YOLO models were developed on a custom framework Darknet which is written 
mainly in C by Alexey Bochkovsky. Ultralystic is the company that converts previous versions of 
YOLO on one of the most famous frameworks in the field of deep learning, PyTorch which is written 
in the Python language.
4.1 Overview of YOLOv5
Besides, Glenn Jocher is also the inventor of the Mosaic data augmentation and acknowledged by 
Alexey Bochkovsky in the YOLOv4 paper (Bochkovskiy, et al., 2020). However, his YOLOv5 model 
caused lots of controversy in the computer vision community because of its name and 
improvements.
Despite being released a month after YOLOv4, the start of research for YOLOv4 and YOLOv5 was 
quite close (March – April 2020). For avoiding collision, Glenn decided to name his version of 
YOLO, YOLOv5. Thus, basically, both researchers applied the state-of-the-art innovations in the 
field of computer vision at that time. That makes the architecture of YOLOv4 and YOLOv5 very 
similar and it makes many people dissatisfied with the name YOLOv5 (5th generation of YOLO) 
when it does not contain multiple outstanding improvements compared to the previous version 
YOLOv4. Besides, Glenn did not publish any paper for YOLOv5, causing more suspicions about 
YOLOv5.
However, YOLOv5 possessed the advantages in engineering. YOLOv5 is written in Python 
programming language instead of C as in previous versions. That makes installation and integration 
on IoT devices easier. In addition, the PyTorch community is also larger than the Darknet 
community, which means that PyTorch will receive more contributions and growth potential in the 
future. Due to being written in 2 different languages on 2 different frameworks, comparing the 
performance between YOLOv4 and YOLOv5 is difficult to be accurate. But after a while, YOLOv5 
37
has proved higher performance than YOLOv4 under certain circumstances and partly gained 
confidence in the computer vision community besides YOLOv4.
4.2 Notable differences – Adaptive anchor boxes
As mentioned above, the YOLOv5 architecture has integrated the latest innovations similar to the 
YOLOv4 architecture, thus there are not many brilliant differences in theory. The author did not 
publish a detailed paper, but only launched a repository on Github and updates improvements 
there. By dissecting its structure code in file .yaml, the YOLOv5 model can be summarized as 
follows (Jocher, 2020):
- Backbone: Focus structure, CSP network
- Neck: SPP block, PANet
- Head: YOLOv3 head using GIoU-loss
The remarkable point mentioned by the YOLOv5 author is an engineering difference. Joseph 
Redmon introduced the anchor box structure in YOLOv2 and a procedure for selecting anchor 
boxes of size and shape that closely resemble the ground truth bounding boxes in the training set. 
By using the k-mean clustering algorithm with different 𝑘 values, the authors picked the 5 best-fit 
anchor boxes for the COCO dataset (containing 80 classes) and use them as the default. That 
reduces training time and increases the accuracy of the network.
However, when applying these 5 anchor boxes to a unique dataset (containing a class not belonged 
to 80 classes in the COCO dataset), these anchor boxes cannot quickly adapt to the ground truth 
bounding boxes of this unique dataset. For example, a giraffe dataset prefers the anchor boxes 
with the shape thin and higher than a square box. To address this problem, computer vision 
engineers usually run the k-mean clustering algorithm on the unique dataset to get the best-fit 
anchor boxes for the data first. Then, these parameters will be configurated manually in the YOLO 
architecture.
Glenn Jocher proposed integrating the anchor box selection process into YOLOv5. As a result, the 
network has not to consider any of the datasets to be used as input, it will automatically "learning" 
the best anchor boxes for that dataset and use them during training. (Solawetz, 2020)
"

**Comment by Keiku** [comment]
Where is the yolov5 paper written by "glenn-jocher"? This is the correct issue. Everyone is waiting for it because he commented on it himself.

**Comment by dinhanhx** [comment]
21 days till 2022, still no paper?

**Comment by MaKaNu** [comment]
@glenn-jocher I prepared it for you:
![Tk0uEdf9OPQuQFghgiBDaBm-J5hA3btAcVrZhckgUtQ](https://user-images.githubusercontent.com/32844273/145605956-c10d9522-c42b-4f76-9265-ca31fad8d43a.png)

Sorry I had to do it ;)

**Comment by Makki93** [comment]
I don't want to stir up hatred here, but I have the bad feeling that YOLOv5 just doesn't perform as well as wanted to release a paper with results in a general comparison. Or they just don't do anything majorly different to v4 so it would justify a scientific paper.

I understand if what I stated above is true, that it's a strategy to not release a paper yet. Ultralytics can't afford any bad reputation when they want to earn money in the future with YoloV5. But I don't get the point of making promises that aren't kept or even answered anymore.

They could just say they're only focussing on YoloV5 being a well (not superior) performing object detector which is super easy to integrate everywhere, as they already do. Nothing more.

**Comment by Makki93** [comment]
Here's a benchmark image from the YOLOv4 author, published over pjreddie's darknet repository (YOLOv1-3), to underpin my statement:

![image](https://user-images.githubusercontent.com/44347047/147065846-cf6c48e5-1ba3-4a55-97d7-551b992d1f6d.png)

source: https://github.com/pjreddie/darknet

**Comment by maxzuo** [comment]
@Makki93 That's a great chart for comparisons. I generally agree with the sentiment of your statement, and I think that's very plausible.

But one thing that doesn't really make sense for me is when you incorporate GPU latency/fps. Do you have any idea how to compare the speeds of these models? On the ultralytics page they report the YOLOv5x6 model having less than 20ms latency on the V100 whereas comparably the YOLOv4-P6 needs more than 30ms, and the reported YOLOv4-P7 above requires more than 60ms.

Maybe they have something there with regards to the speed of their model?

And the fact they are likely evaluating at different image sizes doesn't help make the comparison easier.

**Comment by Alzubair98** [comment]
> 21 days till 2022, still no paper?

only few hours now, still no paper

**Comment by dinhanhx** [comment]
> > 21 days till 2022, still no paper?
> 
> only few hours now, still no paper

And it's already 2022 in Japan

**Comment by ssooffiiaannee** [comment]
No paper yet? how can we know the difference between yolov5s yolov5x yolov5l and others? I cannot run these models without knowing what they are doing, no need for a paper, just a small article.
@glenn-jocher

**Comment by darkC00kie** [comment]
I would also be really interested in reading a paper about the architecture and yolo in general as I am writing my thesis on this subject.

@glenn-jocher

**Comment by AyadAlmamary** [comment]
Hi, while we are waiting for Yolo-v5 paper, I want to test my dataset on Yolo3 and v4, anyone has a Colab file to do the training on custom dataset.
thanks

**Comment by besmaguesmi** [comment]
@AyadAlmamary  https://medium.com/analytics-vidhya/train-a-custom-yolov4-object-detector-using-google-colab-61a659d4868

**Comment by AyadAlmamary** [comment]
@besmaGuesmi You are here also , Thanks a lot.

**Comment by vikxoxo** [comment]
You may also search for Roboflow tutorials for yolo v3 & 4

**Comment by AyadAlmamary** [comment]
@glenn-jocher 
Hi, 
we are waiting for YOLO-v5 paper, we want to cite it in our researches paper, Any update about it?
-Also, I want a way to contact with you, email or something else?

**Comment by Vishal-rkavk** [comment]
any update on YOLOV5 paper?  @glenn-jocher

**Comment by ssooffiiaannee** [comment]
@AyadAlmamary  his email : glenn.jocher@ultralytics.com, I wrote him, he has answered my questions except the one about the paper. If there is no paper, please just say it @glenn-jocher , I think avoiding all questions about the paper will lead to distrust.

**Comment by BanzerUniKL** [comment]
I actually have to use a different model for my master-thesis if there won't come a paper. please give us informations about the state; if there won't come it is fine, i just need to know

**Comment by MarigoldJ** [comment]
I just got here for searching YOLO paper, but I found that only v5 paper doesn't exist. :(

**Comment by kangsan0420** [comment]
so, how did your hat taste?

**Comment by thaingoc2604** [comment]
Can I ask if there is an article about yolov5?

**Comment by Lininggggggg** [comment]
请问没论文怎么引用啊，有哥们能提供引用格式吗？

**Comment by zldrobit** [comment]
@Lininggggggg How about using [DOI ](https://zenodo.org/badge/latestdoi/264818686)for a citation?

**Comment by Lininggggggg** [comment]
oh, nice, thank you!

**Comment by Lininggggggg** [comment]
> @Lininggggggg How about using [DOI ](https://zenodo.org/badge/latestdoi/264818686)for a citation?

oh, nice, thank you!

**Comment by thaingoc2604** [comment]
I train model yolov5s (.pt) now I want to convert it to .tflite, what should I do?

**Comment by zldrobit** [comment]
@thaingoc2604 Plz run 
```
python export.py --weights path/to/yolov5s.pt --include tflite
```
, and refer to https://docs.ultralytics.com/yolov5/tutorials/model_export .

**Comment by pcruiher08** [comment]
So where's the paper?

**Comment by hupf80** [comment]
Also searching for a Paper of Yolov5 for my uni-research ...

**Comment by ghost** [comment]
![6k4ndb](https://user-images.githubusercontent.com/35607198/174458880-ec54f123-462c-40e1-9a8a-b652aa9fbc95.jpg)

**Comment by gogoymh** [comment]
Is there any document for how and why architecture is built like that?

**Comment by serifdogruu** [comment]
still not there ?

**Comment by Keiku** [comment]
The YOLOv7 paper has been released. Is the YOLOv5 paper yet?

[2207.02696] YOLOv7: Trainable bag-of-freebies sets new state-of-the-art for real-time object detectors https://arxiv.org/abs/2207.02696

**Comment by abuod0** [comment]
Anything yet on Yolov5 papers?

**Comment by eliabntt** [comment]
2023 😂

**Comment by abuod0** [comment]
I guess we'd never see it, similarly for ChatGPT..

**Comment by harshdhamecha** [comment]
They released YOLOV8. Only God knows whether v8 paper would come or not

**Comment by Suzan009** [comment]
> @besmaGuesmi we are targeting a paper release by the start of PyTorch Dev day, December 1st.
> 
> If the YOLOv5 paper is not published by then I will eat my hat.

Now, such your d*ck.

**Comment by ranjanj1** [comment]
March 18 2023, Do we have yolov5 paper yet ?

**Comment by capsule2077** [comment]
March 26 2023, Do we have yolov5 paper yet ? yolov7 was  released  in cvpr2023,i think yolov5 will be released  in cvpr???? .

**Comment by erwinschrodinger1** [comment]
June 2023 still no paper?

**Comment by ksasso1028** [comment]
eat yo socks

**Comment by aymuos15** [comment]
https://github.com/ultralytics/ultralytics/issues/204

**Comment by glenn-jocher** [comment]
@aymuos15 hi there,

Thank you for reaching out and for your interest in YOLOv5. I understand your excitement and curiosity regarding the YOLOv5 paper. As of now, there hasn't been an official YOLOv5 paper released. However, please note that the YOLOv5 repository at Ultralytics is an open-source project developed by the Ultralytics team.

The absence of a specific paper does not diminish the value and capabilities of YOLOv5. The repository provides extensive documentation, code, and tutorials, allowing users to understand and utilize the YOLOv5 model effectively. Additionally, the YOLO (You Only Look Once) object detection algorithm has a strong community presence, and the Ultralytics team is committed to maintaining and improving the YOLOv5 repository.

If and when a YOLOv5 paper is released, it would be an exciting development for the YOLO community. Until then, I encourage you to explore the YOLOv5 repository and leverage its features for your object detection tasks.

Please feel free to ask any further questions or share any concerns you may have.

Best regards,

Glenn Jocher

**Comment by charecktowa** [comment]
So no paper in 2024?

**Comment by glenn-jocher** [comment]
@charecktowa hi there! 😊 As of now, a dedicated YOLOv5 paper has not been released. The good news is, the YOLOv5 GitHub repo is rich with detailed documentation, tutorials, and updates to support your projects. While we understand the interest in a formal paper, the repo and our active community are your go-to resources for everything YOLOv5. Thank you for your enthusiasm and support!

**Comment by aminahamdi5** [comment]
14/02/2025, Anything yet about Yolov5 paper?

**Comment by pderrenger** [comment]
Hi there! As of now, there is no official YOLOv5 paper. However, you can explore the detailed [YOLOv5 documentation](https://docs.ultralytics.com/yolov5) and [GitHub repository](https://github.com/ultralytics/yolov5) for implementation details, architecture insights, and community contributions. The Ultralytics team focuses on practical advancements and open-source resources to support users.

**Comment by EnriqueMoran** [comment]
So no YOLOv5 paper?

**Comment by pderrenger** [comment]
Hi @EnriqueMoran—correct, there’s no official YOLOv5 paper as of November 4, 2025; for academic use please cite the repository’s Citations section and refer to the Ultralytics Docs for details. See the Ultralytics Docs.

**Comment by ParthProLegend** [comment]
Hello, Good day to you. I was going through your work and the papers.

<img width="1774" height="282" alt="Image" src="https://github.com/user-attachments/assets/d9a8c86c-f88b-45f7-a42b-a79e93f48fae" />

Your Abstract at https://explore.openaire.eu/search/result?pid=10.5281%2Fzenodo.7347926 has html in it, is this intended?

**Comment by pderrenger** [comment]
Hi @ParthProLegend, good catch—no, that HTML in the abstract isn’t intentional; it’s coming from formatted text in the underlying record and not being fully sanitized in the index, and I’ll flag this to the team so we can clean up the metadata in a future update.
